CDCgov/prime-reportstream
# 6634 remove name property from FHIRtoHL7 conversion

This PR gets rid of the `name` field from configuration files. This field was primarily used for debugging, and is now replaced with references that use the file's name.

Test Steps:
1. Run the converter with the `fhirdata` command. e.g. `./prime fhirdata --input-file src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml --output-format=HL7`.
2. Verify the output is valid HL7.
3. Check the debug output and verify it's readable.

## Changes
- Remove enforcing of a `name` field for the conversion config files
- Remove all existing uses of the `name` field
- Change logger calls in the changed methods to still be readable

## Linked Issues
- #6634
